opm/common/utility/Serializer.hpp

```diff
@@ -545,53 +549,47 @@ class Serializer {
         T, std::void_t<decltype(std::declval<T>().serializeOp(std::declval<Serializer<Packer>&>()))>
     > : public std::true_type {};
 
-    //! \brief Handler for smart pointers.
+    //! \brief Handler for shared pointers.
     template<class PtrType>
-    void ptr(const PtrType& data)
+    void shared_ptr(const PtrType& data)
     {
         using T1 = typename PtrType::element_type;
-        void* data_ptr = reinterpret_cast<void*>(data.get());
+        uintptr_t data_ptr = reinterpret_cast<uintptr_t>(data.get());
         (*this)(data_ptr);
         if (!data_ptr)
             return;
-
         if (m_op == Operation::PACK || m_op == Operation::PACKSIZE) {
             if (m_ptrmap.count(data_ptr) == 0) {
                 (*this)(*data);
-                m_ptrmap[data_ptr] = nullptr;
+                m_ptrmap[data_ptr] = 0;
```

Comment:
I think I'd prefer `m_ptrmap[data_ptr].reset()` here, or maybe just `m_ptrmap.insert_or_assign(data_ptr, std::make_shared<void>())`.

Reply:
i used the first one. would have liked the second one but std::make_shared can't be used for a void pointer